controls/microsoft-365-foundations-7.2.5.rb

```diff
@@ -38,21 +38,13 @@
   ref 'https://learn.microsoft.com/en-us/sharepoint/external-sharing-overview'
 
   ensure_spo_guest_users_cannot_share_items_dont_own_script = %{
-    $client_id = '#{input('client_id')}'
-    $tenantid = '#{input('tenant_id')}'
-    $clientSecret = '#{input('client_secret')}'
-    $certificate_password = '#{input('certificate_password')}'
-    $certificate_path = '#{input('certificate_path')}'
-    $sharepoint_admin_url = '#{input('sharepoint_admin_url')}'
-    Install-Module -Name PnP.PowerShell -Force -AllowClobber
-    import-module pnp.powershell
-    $password = (ConvertTo-SecureString -AsPlainText $certificate_password -Force)
-    Connect-PnPOnline -Url $sharepoint_admin_url -ClientId $client_id -CertificatePath $certificate_path -CertificatePassword $password  -Tenant $tenantid
 	  (Get-PnPTenant).PreventExternalUsersFromResharing
```

Comment:
So my approach here would have been to just call `Get-PnPTenant`, return some data from the teams pnp instance, and then we'd have a nice data structure we could interrogate _in Ruby, inside our control file,_ instead of farming all of that out to a Powershell script.

We could do something like:
```
powershell_output = pwsh_single_session_executor(ensure_spo_guest_users_cannot_share_items_dont_own_script).run_script_in_teams_pnp
raise Inspec::Error, "The powershell output returned the following error:  #{powershell_output.stderr}" if powershell_output.exit_status != 0

describe 'Teams PNP' do
  it 'should prevent external users from resharing' do
    expect(powershell_output['PreventExternalUsersFromResharing']).to cmp true
  end
end
```
I'm not sure exactly what data structure gets returned by the pwsh executor so the syntax might be a bit different but you get the idea. See how we're leveraging RSpec syntax a little better now?

You have in general farmed out a lot of the actual test logic to the powershell scripts, which is not wrong, but is against the pattern we usually use when approaching InSpec queries, which is to a) fetch data from the target and then b) slice and dice that data on the Ruby side to see if it passes our test.